fdbrpc/FlowTests.actor.cpp

```diff
@@ -50,6 +50,26 @@ TEST_CASE("/flow/actorcompiler/lineNumbers") {
 	return Void();
 }
 
+TEST_CASE("/flow/delayOrdering") {
+	state double x = deterministicRandom()->random01();
+	state double y = deterministicRandom()->random01();
+	if (BUGGIFY) {
+		y = x;
+	}
+	state int last = 0;
+	state Future<Void> f1 = map(delay(x), [last = &last](const Void&) {
```

Comment:
```suggestion
	state Future<Void> f1 = map(delay(x), [&last](const Void&) {

```

Reply:
I wish we could do that :) It's structured weirdly to work around flow weirdness
```
$ ninja -C ~/build/foundationdb fdbrpc/CMakeFiles/fdbrpc.dir/FlowTests.actor.g.cpp.o
ninja: Entering directory `/Users/anoyes/build/foundationdb'
[3/3] Building CXX object fdbrpc/CMakeFiles/fdbrpc.dir/FlowTests.actor.g.cpp.o
FAILED: fdbrpc/CMakeFiles/fdbrpc.dir/FlowTests.actor.g.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_SYSTEM_NO_DEPRECATED -DNO_INTELLISENSE -DTLS_DISABLED -DUSE_UCONTEXT -DWITH_LIBCXX -DFNAME=fdbrpc_FlowTests -I/Users/anoyes/workspace/foundationdb -I. -I/Users/anoyes/workspace/foundationdb/fdbrpc -Ifdbrpc -I/Users/anoyes/workspace/foundationdb/fdbrpc/libeio -I/Users/anoyes/workspace/foundationdb/flow -Iflow -IboostProject-prefix/src/boostProject
-g   -DCMAKE_BUILD -ggdb -fno-omit-frame-pointer -maes -mmmx -mavx -msse4.2 -std=c++17 -stdlib=libc++ -Wno-unknown-warning-option -Wno-dangling-else -Wno-sign-compare -Wno-comment -Wno-unknown-pragmas -Wno-delete-non-virtual-dtor -Wno-undefined-var-template -Wno-tautological-pointer-compare -Wno-format -Werror -Wno-error=format -Wunused-variable -Wno-deprecated -fvisibility=hidden -Wreturn-type -fPIC -MD -MT fdbrpc/CMakeFiles/fdbrpc.dir/FlowTests.actor.g.cpp.o -MF fdbrpc/CMakeFiles/fdbrpc.dir/FlowTests.actor.g.cpp.o.d -o fdbrpc/CMakeFiles/fdbrpc.dir/FlowTests.actor.g.cpp.o -c fdbrpc/FlowTests.actor.g.cpp
/Users/anoyes/workspace/foundationdb/fdbrpc/FlowTests.actor.cpp:60:25: error: 'last' in capture list does not name a variable
                        f1 = map(delay(x), [&last](const Void&) { last = 1; return Void(); });
                                             ^
/Users/anoyes/workspace/foundationdb/fdbrpc/FlowTests.actor.cpp:60:46: error: 'this' cannot be implicitly captured in this context
                        f1 = map(delay(x), [&last](const Void&) { last = 1; return Void(); });
                                                                  ^
/Users/anoyes/workspace/foundationdb/fdbrpc/FlowTests.actor.cpp:64:25: error: 'last' in capture list does not name a variable
                        f2 = map(delay(y), [&last](const Void&) { last = 2; return Void(); });
                                             ^
/Users/anoyes/workspace/foundationdb/fdbrpc/FlowTests.actor.cpp:64:46: error: 'this' cannot be implicitly captured in this context
                        f2 = map(delay(y), [&last](const Void&) { last = 2; return Void(); });
                                                                  ^
/Users/anoyes/workspace/foundationdb/fdbrpc/FlowTests.actor.cpp:60:46: error: 'this' cannot be implicitly captured in this context
                        f1 = map(delay(x), [&last](const Void&) { last = 1; return Void(); });
                                                                  ^
/Users/anoyes/build/foundationdb/fdbrpc/FlowTests.actor.g.cpp:474:9: note: in instantiation of member function '(anonymous namespace)::FlowTestCase53ActorState<(anonymous namespace)::FlowTestCase53Actor>::a_body1' requested here
                this->a_body1();
                      ^
/Users/anoyes/workspace/foundationdb/fdbrpc/FlowTests.actor.cpp:64:46: error: 'this' cannot be implicitly captured in this context
                        f2 = map(delay(y), [&last](const Void&) { last = 2; return Void(); });
                                                                  ^
6 errors generated.
ninja: build stopped: subcommand failed.
```

Reply:
I see. The actor compiled code is strange. I tested that
`	state Future<Void> f1 = map(delay(x), [this](const Void&) {`
works. But such a change is also a bit weird, since "this" is not clear either. So I guess your style is better.